std-bits/table/src/main/java/org/enso/table/data/column/operation/cast/ToBigIntegerConverter.java

```diff
@@ -77,6 +80,22 @@ private Storage<BigInteger> convertBoolStorage(
     return builder.seal();
   }
 
+  private Storage<BigInteger> convertBigDecimalStorage(
+      BigDecimalStorage bigDecimalStorage, CastProblemAggregator problemAggregator) {
+    int n = bigDecimalStorage.size();
+    BigIntegerBuilder builder = new BigIntegerBuilder(n, problemAggregator);
+    for (int i = 0; i < n; i++) {
+      BigDecimal value = bigDecimalStorage.getItemBoxed(i);
+      if (value == null) {
+        builder.appendNulls(1);
+      } else {
+        BigInteger bigInteger = value.toBigInteger();
+        builder.appendRawNoGrow(bigInteger);
+      }
+    }
```

Comment:
We probably want a `context.safepoint()` in this loop as well

Reply:
Done